File: leetcode/RectangleArea.java
public class RectangleArea {
  public int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
    long area = (long)(C - A) * (D - B);
    area += (long)(G - E) * (H - F);
    long lx = Math.max(A, E);
    long ly = Math.max(B, F);
    long hx = Math.min(C, G);
    long hy = Math.min(D, H);
    if (lx <= hx && ly <= hy) {
      long area1 = (long)(hx - lx) * (hy - ly);
      area -= area1;
    }
    return (int)area;
  }
}

File: leetcode/RectangleArea.cpp
class Solution {
public:
    int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
        int a1 = (C - A) * (D - B);
        int a2 = (G - E) * (H - F);
        int sub = 0;
        int left = max(A, E);
        int right = min(C, G);
        int bottom = max(B, F);
        int top = min(D, H);
        if (left < right && bottom < top) {
            sub = (right - left) * (top - bottom);
        }
        return a1 - sub + a2;
    }
};
